RUFAS/biophysical/animal/herd_manager.py

```diff
@@ -1,2 +1,2079 @@
+        self.herd_statistics.avg_calving_interval = (
+            sum([cow.reproduction.calving_interval for cow in self.cows]) / len(self.cows) if len(self.cows) > 0 else 0
+        )
+
+    def _update_heifer_reproduction_statistics(self) -> None:
+        """
+        Updates the reproduction statistics of heifers in the herd.
+
+        returns
+        ------
+        None
+
+        """
+        self.herd_statistics.GnRH_injection_num_h = sum(
+            [heiferII.reproduction.reproduction_statistics.GnRH_injections for heiferII in self.heiferIIs]
+        )
+        self.herd_statistics.PGF_injection_num_h = sum(
+            [heiferII.reproduction.reproduction_statistics.PGF_injections for heiferII in self.heiferIIs]
+        )
+        self.herd_statistics.CIDR_count = sum(
+            [heiferII.reproduction.reproduction_statistics.CIDR_injections for heiferII in self.heiferIIs]
+        )
+        self.herd_statistics.preg_check_num_h = sum(
+            [heiferII.reproduction.reproduction_statistics.pregnancy_diagnoses for heiferII in self.heiferIIs]
+        )
+        self.herd_statistics.semen_num_h = sum(
+            [heiferII.reproduction.reproduction_statistics.semen_number for heiferII in self.heiferIIs]
+        )
+        self.herd_statistics.ai_num_h = sum(
+            [heiferII.reproduction.reproduction_statistics.AI_times for heiferII in self.heiferIIs]
+        )
+        self.herd_statistics.ed_period_h = sum(
+            [heiferII.reproduction.reproduction_statistics.ED_days for heiferII in self.heiferIIs]
+        )
+        self.herd_statistics.avg_breeding_to_preg_time = (
+            sum([heiferII.reproduction.breeding_to_preg_time for heiferII in self.heiferIIs if heiferII.is_pregnant])
+            / len(self.heiferIIs)
+            if len(self.heiferIIs) > 0
+            else 0
+        )
+
+    def _update_average_mature_body_weight(self) -> None:
+        """
+        Updates the average mature body weight of the animals in the herd.
+
+        Returns
+        -------
+        None
+
+        """
+        all_animals: list[Animal] = self.calves + self.heiferIs + self.heiferIIs + self.heiferIIIs + self.cows
+        self.herd_statistics.avg_mature_body_weight = (
+            sum([animal.mature_body_weight for animal in all_animals]) / len(all_animals) if len(all_animals) > 0 else 0
+        )
+
+    def _update_average_cow_body_weight(self) -> None:
+        """
+        Updates the average body weight of cows in the herd.
+
+        Returns
+        -------
+        None
+
+        """
+        self.herd_statistics.avg_cow_body_weight = (
+            sum([cow.body_weight for cow in self.cows]) / len(self.cows) if len(self.cows) > 0 else 0
+        )
+
+    def _update_average_cow_parity(self) -> None:
+        """
+        Updates the average cow parity number in the herd statistics.
+
+        Returns
+        -------
+        None
+
+        """
```

Comment:
```suggestion
        """Updates the average cow parity number in the herd statistics."""
```
Removing the "Returns ... None" from the docstrings would shave some lines off this file, which would add up to be pretty beneficial IMO.